packages/Tag/src/Tag.styles.js

```diff
@@ -0,0 +1,158 @@
+import styled, { css } from "styled-components";
+import tokens from "@paprika/tokens";
+import RawButton from "@paprika/raw-button";
+import { fontSize } from "@paprika/stylers/lib/helpers";
+import { truncateText } from "@paprika/stylers/lib/includes";
+
+const tagColorStyles = {
+  black: css`
```

Comment:
Should probably be using the `types` for these colour keys

Reply:
@mikrotron What do you think about naming these "themes" instead? Most of them set not only the `background-color`, but also the `color` for the text. For the "alert" theme it could also add the `border-color` as part of the theme...

Reply:
I think you've got a good point, I'm a little worried that the actual term `theme` comes with some preconceptions... 🤔 But since a bunch of the `color` options are actually risk levels, and it would make a lot of sense to include `border-color` in the same prop. 

Unless @sunminsarahlee has a better suggestion though, I think `theme` is better than `tagColor` for now.   If we decide to rename it we'll just have to bump the version 🤷‍♂️

Reply:
[Blueprint.js](https://blueprintjs.com/docs/) uses `intent` throughout their design system instead of `color`. But that's just one example.

Reply:
im good with `theme`

Reply:
This is my proposal. 
```jsx
// Pill component
Tags.theme = (...args) => {
  const [color=null, background=null, border=null] = args;
  let _color = "black";
  let _backround = "white";
  let _border: "1px solid transparent";
  
  if(color) _color = color
  if(background) _background = background;
  if(border) _border = border;
  
  return css`${color}; ${background}; ${border}`;
}
Tags.types.kind = { 
  danger: Tags.theme(["red", "white", "red"]), 
  success: Tags.theme(["white", "green"]) 
};
```

// 
```jsx
<Tag kind={Tags.types.kind.success} />
<Tag kind={Tags.theme(["blue", "white", "blue"])} />
```

With an approach like this, the component could still be behaving similar to Button component with the prop `kind` but will also have a powerful function that lets you theme some part of the component, without breaking the size and the main design.

@mikrotron @tristanjasper

Reply:
`kind` can have of course their default value. and the css value that you get from `kind` prop can be spread directly in the styled component.

Reply:
I like where this suggestion is going, but have some suggestions about the API design.  For now, I think it's okay to proceed with the simple `theme` prop as @tristanjasper has implemented so he can get this merged.  We can bump to `v2.0` if we like this kind of API for Paprika, and start using it more consistently throughout the library.

Reply:
Suggestions:

- Rename `kind` → `theme` and `theme` → `themeGenerator` (or something better)
- Use an object for the `args` instead of an array
- Consider how we can document this API (specifically the shape of those `args` for `themeGenerator`)